src/useInterval/useInterval.test.ts

```diff
@@ -0,0 +1,85 @@
+import { runComponentSetup } from '@muban/test-utils';
+import { timeout } from '../useTimeout/useTimeout.test.utils';
+import { useInterval } from './useInterval';
+
+jest.mock('@muban/muban', () => jest.requireActual('@muban/test-utils').getMubanLifecycleMock());
+
+describe('useInterval', () => {
+  it('should not crash', async () => {
+    await runComponentSetup(() => {
+      useInterval(() => undefined);
+    });
+  });
+
+  it('should start immediate and not be completed', async () => {
+    const mockHandler = jest.fn();
+
+    await runComponentSetup(() => {
+      useInterval(mockHandler, 100);
+    });
+
+    expect(mockHandler).toBeCalledTimes(0);
+  });
+
+  it('should start immediate and be called once', async () => {
+    const mockHandler = jest.fn();
+
+    await runComponentSetup(
+      () => useInterval(mockHandler, 100),
+      async ({ stopInterval }) => {
+        await timeout(100);
+        stopInterval();
+      },
+    );
+
+    expect(mockHandler).toBeCalledTimes(1);
+  });
+
+  it('should trigger start and be stopped after three calls', async () => {
+    const mockHandler = jest.fn();
+
+    await runComponentSetup(
+      () => useInterval(mockHandler, 100, false),
+      async ({ startInterval, stopInterval }) => {
+        startInterval();
+        await timeout(400);
+        stopInterval();
+      },
+    );
+
+    expect(mockHandler).toBeCalledTimes(3);
+  });
+
+  it('should trigger stop once the interval is started', async () => {
+    const mockHandler = jest.fn();
+
+    await runComponentSetup(
+      () => useInterval(mockHandler, 200, false),
+      async ({ startInterval, stopInterval }) => {
+        startInterval();
+        await timeout(100);
+        stopInterval();
```

Comment:
wonder if this test also succeeds if we remove the `stop`? Since it takes another 100ms before the interval fires.

It might behave differently if we were to use `useFakeTimers` (https://jestjs.io/docs/timer-mocks), which I think we should to make tests run quicker.

Either way, it might be better to do another `await timeout(200);` after the `stopInterval` to make sure enough time has passed, so we can determine the stop actually works.